src/main/java/slimeknights/tconstruct/plugin/jei/casting/CastingVariantCategory.java

```diff
@@ -0,0 +1,19 @@
+package slimeknights.tconstruct.plugin.jei.casting;
+
+import mezz.jei.api.gui.drawable.IDrawableAnimated;
+import mezz.jei.api.gui.drawable.IDrawableStatic;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+
+public abstract class CastingVariantCategory<T> implements IRecipeCategory<T> {
```

Comment:
Merge this into AbstractCastingCategory, I have no idea why mezz made so many subclasses in the furnace.